src/test/java/com/google/enterprise/cloudsearch/fs/WindowsFileDelegateTest.java

```diff
@@ -865,29 +873,39 @@ public void testMonitorRenameFile() throws Exception {
     Path file = newTempFile("test.txt");
     Path newFile = file.resolveSibling("newName.txt");
     delegate.startMonitorPath(tempRoot, mockEventPusher);
+    mockEventPusher.respond(newDeleteRecord(file), newOperation(file));
+    mockEventPusher.respond(newRecord(newFile), newItem(newFile));
     Files.move(file, newFile, StandardCopyOption.ATOMIC_MOVE);
     // Renaming a file shows up as a change to its old name, its new name.
     checkForChanges(Sets.newHashSet(newDeleteRecord(file), newRecord(newFile)));
   }
 
   @Test
-  public void testMonitorMoveAccrossDirs() throws Exception {
+  public void testMonitorMoveAcrossDirs() throws Exception {
     Path dir1 = newTempDir("dir1");
     Path dir2 = newTempDir("dir2");
     Path file1 = newTempFile(dir1, "test.txt");
     Path file2 = dir2.resolve(file1.getFileName());
+    mockEventPusher.respond(newDeleteRecord(file1), newOperation(file1));
+    mockEventPusher.respond(newRecord(file2), newItem(file2));
+    mockEventPusher.respond(newRecord(dir1), newItem(dir1));
+    mockEventPusher.respond(newRecord(dir2), newItem(dir2));
+
     delegate.startMonitorPath(tempRoot, mockEventPusher);
     Files.move(file1, file2);
     // Moving a file shows up as a change to its old name, its new name,
     // its old parent, and its new parent.
     checkForChanges(
         Sets.newHashSet(
-            newDeleteRecord(file1), newRecord(file2), newRecord(dir1), newRecord(dir2)));
+            newDeleteRecord(file1), newRecord(file2), newRecord(dir1), newRecord(dir2)),
+        Optional.of(Sets.newHashSet(
```

Comment:
Is this just more monitor weirdness on different versions of Windows?

Reply:
Yeah, it can't seem to make up its mind.